### ISSUE TITLE
Extracting parts of a datetime in a query

### ISSUE THREAD

**Issue by mhauru** [opendp/smartnoise-sdk#542]
I have a datetime column from which I would like to extract the year part. I've tried a few different ways, all of which are valid (Postgre)SQL queries, but they all run afoul of SmartNoise SQL's parser:

```
SELECT count(*) AS num, col_a, DATE_PART('year', col_b) AS col_b_year FROM schema_name.table_name GROUP BY col_a, col_b_year
```
Errors out with `ValueError: Bad token ( at line 1 column 40.  Message: missing FROM at '('`

```
SELECT count(*) AS num, col_a, EXTRACT(YEAR FROM col_b) AS col_b_year FROM schema_name.table_name GROUP BY col_a, col_b_year
```
Errors out with `AttributeError: 'str' object has no attribute 'children'`, coming from `snsql/xpath/ast.py`, line 222.

Would you have any advice on how to do this so that SmartNoise SQL would be able to parse the query? Thanks!

EDIT: The error message in the second case has been edited, the first version was an error on my side.

**Comment by joshua-oss** [comment]
Thanks very much for reporting this issue.  It was indeed a bug in EXTRACT where symbol resolution made a copy with datepart and target expression swapped.  All of our unit tests for these functions are run on ASTs with no symbols loaded, so we totally missed this.  It should be fixed now in smartnoise-sql==0.2.11.  Note that you will need to use syntax like:

```python
query = SELECT EXTRACT(YEAR FROM col_b) AS col_b_year FROM schema_name.table_name GROUP BY EXTRACT(YEAR FROM col_b)'
```

This is because the EXTRACT in the GROUP BY is executed in the database engine, while the EXTRACT in the SELECT is run inside SmartNoise, as post-processing over the differentially private aggregates.  Please let me know if the issue persists in the new release.

**Comment by mhauru** [comment]
Works splendidly in 0.2.11, thanks @joshua-oss!
